python/bindings/include/openravepy/openravepy_int.h

```diff
@@ -331,18 +331,42 @@ inline RaveTransformMatrix<T> ExtractTransformMatrixType(const py::object& o)
 
 inline py::object toPyArrayRotation(const TransformMatrix& t)
 {
+#ifdef USE_PYBIND11_PYTHON_BINDINGS
+    const std::array<dReal, 9> arr {
+        t.m[0], t.m[1], t.m[2],
+        t.m[4], t.m[5], t.m[6],
+        t.m[8], t.m[9], t.m[10]
+    };
+    py::array_t<dReal> pyvalues = toPyArray(arr);
+    pyvalues.resize({3, 3});
+    return pyvalues;
+#else // USE_PYBIND11_PYTHON_BINDINGS
     npy_intp dims[] = {3,3};
     PyObject *pyvalues = PyArray_SimpleNew(2,dims, sizeof(dReal)==8 ? PyArray_DOUBLE : PyArray_FLOAT);
     dReal* pdata = (dReal*)PyArray_DATA(pyvalues);
     pdata[0] = t.m[0]; pdata[1] = t.m[1]; pdata[2] = t.m[2];
     pdata[3] = t.m[4]; pdata[4] = t.m[5]; pdata[5] = t.m[6];
     pdata[6] = t.m[8]; pdata[7] = t.m[9]; pdata[8] = t.m[10];
     return py::to_array_astype<dReal>(pyvalues);
+#endif // USE_PYBIND11_PYTHON_BINDINGS
 }
 
 template <typename T>
 inline py::object toPyArray3(const std::vector<RaveVector<T> >& v)
 {
+#ifdef USE_PYBIND11_PYTHON_BINDINGS
+    const size_t numel = v.size() * 3;
+    std::vector<T> arr(numel);
```

Comment:
@tgn3000 is there a way to do this without creating a new `vector`?

Reply:
yes, done; same for the rest